app/services/report_importer.rb

```diff
@@ -2,14 +2,38 @@ class ReportImporter
   delegate :logger, :to => :Rails
   attr_reader :report
 
+  def self.importer_for(type)
```

Comment:
The registration of a type to an importer is unused in the current arrangement, as you've got a controller per type, whereas facts (which this is copied from) had a single controller with the type passed in.